internal/storage/bucket_handle.go

```diff
@@ -64,15 +64,12 @@ func (bh *bucketHandle) NewReader(
 	}
 
 	// Creating a NewRangeReader instance.
-	r, err := obj.NewRangeReader(ctx, start, length)
+	rc, err = obj.NewRangeReader(ctx, start, length)
```

Comment:
We can return directly, no need to create a "rc", "err" variable.

Reply:
Also, please add a one line comment that - "storage.Reader" is a io.ReadCloser, since it contains both Read() and Close() method.